### PR TITLE
support disabling alertmanager and its resources on mimir backed installations

### PR DESCRIPTION
**Pull request by QuentinBisson**
Towards https://github.com/giantswarm/roadmap/issues/3513
Add support to disable alertmanager if monitoring is disabled at the installation level 

## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: service/controller/managementcluster/resource.go [hunk 1]
```diff
@@ -143,6 +144,8 @@ func newResources(config resourcesConfig) ([]resource.Interface, error) {
 			K8sClient: config.K8sClient,
 			Logger:    config.Logger,
 
+			AlertmanagerEnabled: config.AlertmanagerEnabled && config.MimirEnabled,
```
Comment (TheoBrigitte) [inline service/controller/managementcluster/resource.go:147]: Funny indentation

**Review by TheoBrigitte — APPROVED** [review]
LGTM

File: service/controller/managementcluster/resource.go (hunk 1 above)
Comment (QuentinBisson) [reply service/controller/managementcluster/resource.go:147]: That's not an indentation :D I'm making sure this is only on capi

Comment (TheoBrigitte) [reply service/controller/managementcluster/resource.go:147]: Ignore me, that's my screen being small :D

**Comment by QuentinBisson** [comment]
Tested on golem

**Pull request merged**